stl/inc/ranges

```diff
@@ -1960,9 +1960,13 @@ namespace ranges {
             }
 
             _NODISCARD constexpr decltype(auto) operator*() const
-                _NOEXCEPT_IDL0(noexcept(_STD invoke(*_Parent->_Fun, *_Current))) /* strengthened */ {
+                noexcept(noexcept(_STD invoke(*_Parent->_Fun, *_Current))) {
 #if _ITERATOR_DEBUG_LEVEL != 0
-                _Check_dereference();
+                try {
+                    _Check_dereference();
+                } catch (...) {
+                    _STL_VERIFY(false, "cannot dereference end transform_view iterator");
+                }

```

Comment:
In product code, we never use `try`/`catch` directly - it always has to be `_TRY_BEGIN`/`_CATCH_ALL`/`_CATCH_END` in order to respect `_HAS_EXCEPTIONS=0`, e.g.:
https://github.com/microsoft/STL/blob/afe0800cd6bf0d163bf678ad4e1feb3c130d1ec5/stl/inc/future#L473-L479

However, should we be try-catching at all here? I observe that `filter_view` marks `_Check_dereference()` as `noexcept`. Should we simply do the same for `transform_view`, and allow exceptions emitted from `_RANGES end(_Parent->_Range)` to slam into `noexcept`? The implementations are identical aside from error messages:

https://github.com/microsoft/STL/blob/afe0800cd6bf0d163bf678ad4e1feb3c130d1ec5/stl/inc/ranges#L1634-L1639
https://github.com/microsoft/STL/blob/afe0800cd6bf0d163bf678ad4e1feb3c130d1ec5/stl/inc/ranges#L1912-L1917